### PR TITLE
Refactor ES query to support timezones and ISO timestamps

### PR DESCRIPTION
**Pull request by phamann**
We were getting reports of content appearing in users digests on consecutive days. After some investigation it turns out this was due to our Elastic Search query rounding up the greater than filter to the nearest day. Therefore allowing for potential overlap between days.

This refactors the query options to use a ISO timestamp instead of the custom ES date math DSL if a `timezone` option is supplied.

I have also removed the `\d` from the end of the relative date math to ensure it doesn't round to nearest day for other use-cases.

### CODE REVIEW

File: functions/create-email/lib/get-es-search-opts.js [hunk 1]
```diff
@@ -22,11 +23,28 @@ module.exports = (concepts, opts) => {
...
+							hour: parseInt(opts.time, 10)
+						},
+						opts.timezone
+					)
+					.subtract(opts.sinceDays, 'days')
+					.toISOString();
+		} else {
+			// Custom ES "Date Math" DSL:
+			// https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html#ranges-on-dates
+			gte = `now-${opts.sinceDays}d`
```
Comment (debugwand) [inline functions/create-email/lib/get-es-search-opts.js]: original is `now-${opts.sinceDays}d/d`, doesn't is still need `/d` or is that now moved elsewhere?

File: functions/create-email/lib/get-es-search-opts.js [hunk 2]
```diff
@@ -22,11 +23,28 @@ module.exports = (concepts, opts) => {
 	}
 
 	if (opts.sinceDays) {
+
+		let gte;
+
+		if(opts.time && opts.timezone) {
+			gte = moment.tz({
+							hour: parseInt(opts.time, 10)
```
Comment (debugwand) [inline functions/create-email/lib/get-es-search-opts.js]: if you are only passing in hour, will this work if crossing a date boundary?

File: functions/create-email/lib/get-es-search-opts.js (hunk 1 above)
Comment (phamann) [inline functions/create-email/lib/get-es-search-opts.js]: See the PR description:
> I have also removed the /d from the end of the relative date math to ensure it doesn't round to nearest day for other use-cases.

This was the root cause of our issue as the `/d` causes the time to be rounded to nearest day.

File: functions/create-email/lib/get-es-search-opts.js (hunk 2 above)
Comment (phamann) [inline functions/create-email/lib/get-es-search-opts.js]: Yes, I'm assuming `moment-timezone` will account for this.

File: functions/create-email/lib/get-es-search-opts.js (hunk 1 above)
Comment (debugwand) [inline functions/create-email/lib/get-es-search-opts.js]: ah ok, was comparing with notes on the momentjs website

**Comment by debugwand** [comment]
looks good

**Pull request merged**